### ISSUE TITLE
Cache findup results

### ISSUE THREAD

**Issue by nchen63** [palantir/tslint#2703]
see #2681

**Comment by ajafff** [comment]
On second thought, that might cause problems for editor plugins like `vscode-tslint` or `tslint-language-service` that start a linter process once. If the user moves, deletes or creates a tslint.json that process would need to be restarted to clear the cache.

I'd like to hear what @egamma thinks about this.

**Comment by egamma** [comment]
Not clear what the benefit of caching the findup result is unless the cache is also validated.  `vscode-tslint` and the `tslint-languge-service` cache one `findConfiguration()` lookup for the document that is currently being edited by the user. This configuration is then passed to:

``` js
tslint.lint(path, contents, configuration)
```
The cache is validated/flushed as soon as the user switches to another document. I therefore think that vscode-tslint is in a better position to maintain the cache than the tslint core.

**Comment by JoshuaKGoldberg** [comment]
Given the upcoming deprecation of TSLint and move to ESLint (#4534), this smaller performance concern seems out of scope now.
